airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java

```diff
@@ -101,11 +103,22 @@ public AbstractJdbcSource(final String driverClass,
   protected AutoCloseableIterator<JsonNode> queryTableFullRefresh(final JdbcDatabase database,
                                                                   final List<String> columnNames,
                                                                   final String schemaName,
-                                                                  final String tableName) {
+                                                                  final String tableName,
+                                                                  final SyncMode syncMode,
+                                                                  final Optional<String> cursorField) {
     LOGGER.info("Queueing query for table: {}", tableName);
-    return queryTable(database, String.format("SELECT %s FROM %s",
-        enquoteIdentifierList(columnNames, getQuoteString()),
-        getFullyQualifiedTableNameWithQuoting(schemaName, tableName, getQuoteString())));
+    // This corresponds to the initial sync for in INCREMENTAL_MODE. The ordering of the records matters as intermediate state messages are emitted.
+    if (syncMode.equals(SyncMode.INCREMENTAL)) {
+      final String quotedCursorField = enquoteIdentifier(cursorField.get(), getQuoteString());
+      return queryTable(database, String.format("SELECT %s FROM %s ORDER BY %s ASC",
+          enquoteIdentifierList(columnNames, getQuoteString()),
+          getFullyQualifiedTableNameWithQuoting(schemaName, tableName, getQuoteString()), quotedCursorField));
```

Comment:
`quotedCursorField` 
Checking: Is this going to be the correct fully qualified name with correct escaping?

Reply:
Yup - I've added an additional log in `queryTable`. An example for a sample table is `SELECT \"id\",\"name\",\"power\" FROM \"public\".\"id_and_name\" ORDER BY \"id\" ASC"`

It's the same logic that is used in the existing incremental query (where we do sort by cursor) : https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-jdbc/src/main/java/io/airbyte/integrations/source/jdbc/AbstractJdbcSource.java#L305